### PR TITLE
Improve cell decorator

### PR DESCRIPTION
**Pull request by joamatab**
- add post_process to cell
- deprecated routing settings from CrossSection

```python
import gdsfactory as gf


@gf.cell(post_process=[gf.add_padding, gf.add_pins.add_pins])
def straight_sc(**kwargs):
    return gf.components.straight(**kwargs)


if __name__ == "__main__":
    c = straight_sc()
    c.show()
```

this is similar to Troy's PR but in the decorator @tvt173 

@nikosavola

### CODE REVIEW

File: gdsfactory/cell.py [hunk 1]
```diff
@@ -146,6 +123,11 @@ def mzi_with_bend():
         mzi_with_bend_decorated = gf.cell(mzi_with_bend)
 
     """
+    if default_decorator is not None:
+        warnings.warn(
+            "default_decorator is deprecated and will be removed soon. Use post_process instead.",
+            stacklevel=2,
```
Comment (nikosavola) [inline gdsfactory/cell.py:129]: I think we should also use the warning category so one can filtre out deprecation warnings easily if desired
```suggestion
            "default_decorator is deprecated and will be removed soon. Use post_process instead.",
            DeprecationWarning,
            stacklevel=2,
```

File: gdsfactory/cell.py [hunk 2]
```diff
@@ -311,6 +294,11 @@ def wrapper(*args, **kwargs) -> Component:
             component.module = func.__module__
             component.__doc__ = func.__doc__
 
+        for post in post_process or []:
+            component = post(component)
```
Comment (nikosavola) [inline gdsfactory/cell.py:298]: Is this now the same as the `decorator` argument that we had before?

**Review by nikosavola — CHANGES_REQUESTED** [review]

File: gdsfactory/cell.py (hunk 2 above)
Comment (joamatab) [reply gdsfactory/cell.py:299]: yes, but this one is applied as 

```
@gf.cell(post_process=[add_label, add_marker_layer])
def straight_pdk_a(**kwargs)
   return gf.components.straight(**kwargs)

```

**Review by nikosavola — APPROVED** [review]

**Comment by joamatab** [comment]
merged this one instead

https://github.com/gdsfactory/gdsfactory/pull/2610

I can deprecate routing settings from CrossSection on a separate PR

**Pull request closed without merging**